### DISCUSSION TITLE
Your copilot experience is not fully configured, complete your setup.

### DISCUSSION THREAD

**Discussion by GetFookedWeeb** [community/community#19083] [Copilot Conversations] [answered]
I got accepted on the waitlist a while back and finally got a chance to try copilot, I installed the extension but was signed in with a different GibHub account than the one I requested access with. So i obviously didn't have access.
I tried reinstalling the extension, reloading vscode, removing my username and email from the git config through the terminal.
But i still get the message "your copilot experience is not fully configured complete your setup".
![image](https://user-images.githubusercontent.com/46489753/174889470-0facc540-2b28-4aa2-8ad7-359a5434d022.png)
Clicking "copilot settings" just takes me to https://github.com/settings/copilot where it asks me to start a free trial. Not sure why? But even so, starting a new trial doesn't change anything
Copilot output:
`[INFO] [auth] [2022-06-21T20:08:22.512Z] Invalid copilot token: missing token: 403 
[ERROR] [default] [2022-06-21T20:08:22.514Z] GitHub Copilot could not connect to server. Extension activation failed: "User not authorized"
`

**Comment by davecheney** [comment, marked as answer]
<img width="475" alt="Screen Shot 2021-11-30 at 8 03 57 am" src="https://user-images.githubusercontent.com/7171/143942796-0c2b6d34-e0f4-46fe-b349-f26087b2c5ed.png">
Please try signing out, copilot will prompt you to sign back in.

**Reply by wintrov** [reply]
and login and it's co-pilot back to live!

**Reply by domelon-depek** [reply]
it is not working ,please help me！

**Reply by HarshYadav152** [reply]
## not solved my same problem

**Reply by aviraw** [reply]
Signing out and signing in worked for me, make sure to select all the preferences in the settings too.

**Reply by Xamp1e** [reply]
> Signing out and signing in worked for me, make sure to select all the preferences in the settings too.

Works on me, make sure to "enanble" those options

**Reply by surajpetkar** [reply]
Hi, please go to github copilot setting and enable all options and in VS code  sign in back then it will work.

**Reply by cutegitguy** [reply]
You should select all options which is in github to complete your setup

**Reply by flyguythecreator** [reply]
I am having a problem when doing so. When selecting the options during the setup, it reverts my responses to "Select an option" and then gives the error "Select an option below to continue." Creating an infinite loop that I cannot proceed beyond.

**Reply by flyguythecreator** [reply]
the response below "The fix is to select an option for "Suggestions matching public code (duplication detection filter)"" by [Andrew-Shay](https://github.com/Andrew-Shay) solved the issue.

**Reply by mkassa** [reply]
I had similar issue: 
1. [Went to copilot setting on github](https://github.com/settings/copilot/features) and enabled basically all of them. 
2. I then signed out my github account from visual studio code. 
![image](https://github.com/user-attachments/assets/6b5611b9-9af1-4a00-9d47-0602b3d2f93b)
3. sign back in by clicking the github icon on your VS code as shown in the picture and the popup should appear.
![image](https://github.com/user-attachments/assets/09a792bf-3ee7-499d-89ae-5879005b3db2)

**Comment by CrescentR** [comment]
uninstall the extension and install it again may make it

**Reply by rogergcc** [reply]
holysh1t it works thanks dude

**Reply by mminkowicz** [reply]
stil not working

**Reply by domelon-depek** [reply]
it is not working ,please help me！

**Reply by SSP6904** [reply]
> it is not working ,please help me！

It's already fixed by GitHub.

**Comment by Ender-Wang** [comment]
I am suddenly getting this message on the github webpage, vscode extension is normal tho, but also tried sign out there and sign back in, still having this complaint on the web, how do I get it off? I think I am all set.
![CleanShot_2024-09-17_22-08-02](https://github.com/user-attachments/assets/da932351-9c9c-42ce-84fa-8620673c9d43)
![Google Chrome 2024-09-17 22 05 53](https://github.com/user-attachments/assets/664cd92d-b0a2-4f84-bbb1-475eae47502e)

**Reply by drupol** [reply]
Same here.

**Reply by MaKTaiL** [reply]
I'm having the same issue.

**Reply by SSP6904** [reply]
Also having the same issue

**Reply by sminor** [reply]
Same!

**Reply by mhmd-aashik** [reply]
same here

**Reply by xjy486** [reply]
Oh, I also meet this question

**Reply by mminkowicz** [reply]
Same issue here someone please help!

Your copilot experience is not fully configured, complete your setup.
I have tried reinstalling vscode reinstalling extensions signing in and out restarting my computer

**Reply by mminkowicz** [reply]
Same issue here anyone have a fix 
I have tried restarting, resigning in, resetting clearing cache nothing is working.

**Reply by TristonH55** [reply]
same here. anyone manage to work this out

**Reply by a1996k** [reply]
is there a fix for this ?
i still have this issue

**Reply by Andrew-Shay** [reply]
The fix is to select an option for "Suggestions matching public code (duplication detection filter)"

**Reply by opzc35** [reply]
> 解决方法是为“匹配公共代码的建议（重复检测筛选器）”选择一个选项

谢谢你，我成功了

**Comment by joshistoast** [comment]
Ahh, thought it was just me :)

**Comment by Ender-Wang** [comment]
It was gone, GitHub fixed it.

**Comment by mminkowicz** [comment]
Anyone have a fix for this issue

Your Copilot experience is not fully configured, complete your setup. You are currently logged in as mminkowicz.

**Reply by SSP6904** [reply]
It's already fixed

**Comment by Pedromrendeiro** [comment]
Thanks for the help!

**Comment by ShodaiUsami** [comment]
I have the same problem. I don't know how can I do. Please tell me how to solve this problem. 
"Your GitHub Copilot setup is incomplete. Select your preferences below to start using GitHub Copilot."

**Reply by SSP6904** [reply]
GitHub already fixed it

**Comment by kingsimba** [comment]
Same problem here.

![image](https://github.com/user-attachments/assets/a47d9b3e-4ac1-4968-8990-74f1e858d26a)

**Reply by SSP6904** [reply]
It's been fixed by GitHub awhile back, so if you're still getting that error on Visual Studio Code (I'm guessing that's what you're using), you may want to reinstall the GitHub Copilot extension or (if possible) try clearing the cache.

**Comment by SSP6904** [comment]
If anyone is still getting that error message on Visual Studio Code, I suggest you reinstall the GitHub Copilot extension by removing it and installing it again and then try to login back into it using the GitHub login. The error message might be cached, so maybe clearing it can help (if that works). This error was only a bug and only for those who have a problem with their subscription to the service. Please let me know if that works out for you! I'm glad to help out with anyone! 😉

**Reply by WWWKKXX** [reply]
I've reinstalled both VSCode and the extension,but still this kind of error(no matter I tried it on windows or linux)what should I do to deal with this? 
![截图 2024-12-13 21-49-48](https://github.com/user-attachments/assets/8c719b93-2df5-4574-b844-c4c5b1cd0e59)

**Reply by SSP6904** [reply]
> I've reinstalled both VSCode and the extension,but still this kind of error(no matter I tried it on windows or linux)what should I do to deal with this? 
> ![截图 2024-12-13 21-49-48](https://github.com/user-attachments/assets/8c719b93-2df5-4574-b844-c4c5b1cd0e59)
> 

Then I'm not sure how you're still getting the same error. Unless it's a payment or account issue, maybe checking one of those might give you a clue on what's going on with it. Otherwise, I can't figure out why you're getting the error. 😞

**Comment by SayantaniDeb** [comment]
i'm facing the same issue, did anyone get to fix it?

**Comment by Obooman** [comment]
For me, visiting https://github.com/settings/copilot?editor=copilot_1.254.0, github showing the same waning banner;
and completing all the configs(enabling or disabling bing search, Claude, etc), refresh the page, the warning is gone;
switch back to VSCode, sign out, and sign in, it works!

**Reply by otakutyrant** [reply]
It works for me. Good job!

**Reply by basilmariano** [reply]
this works for me jsut edit something in the https://github.com/settings/copilot and refresh and try to refresh the vscode

**Reply by zkjisj** [reply]
It works for me, thanks a lot!

**Comment by dzakup** [comment]
Sorry..byt the Copilot is the only complicated AI. It just give me stupid messsage but i don't want to spend hours finding solutions. I really don't care about this product when you are not able to provide smooth free experience.

**Reply by dzakup** [reply]
Your GitHub Copilot setup is incomplete. Select your preferences below to start using GitHub Copilot.

**Comment by HarshYadav152** [comment]
## Hey guys .. I posted the same question the github support ..
### and they gave answer which is work in my case hope it works for you well.
![g-support](https://github.com/user-attachments/assets/97df0b04-37f3-4b1e-ae2a-65a3e2eb55a8)

**Comment by LewisBai97** [comment]
It may be necessary to make relevant settings in GitHub Copilot. At least, I solved the problem by using this method.

**Reply by zkjisj** [reply]
Thanks for your reply, I have solved this issue after making all relative settings done in GitHub Copilot.

**Comment by Lazy-Program** [comment]
I was able to fix my issue without restarting anything or uninstallation. 
All I did was update following settings and then in VS code signed in and out of the github account. 
![image](https://github.com/user-attachments/assets/38f17f0b-08bd-49ea-ac79-7dac466df1aa)

**Reply by jhuaman1** [reply]
Esa era la respuesta. Bien funciona

**Comment by green-comet** [comment]
![image](https://github.com/user-attachments/assets/51ac384e-4d73-4b73-b9b7-e2fabfb8a5aa)
First make this allowed then make all enabled, it worked for me .
